src/core/List.pm6

```diff
@@ -1489,13 +1489,45 @@ my class List does Iterable does Positional { # declared in BOOTSTRAP
     }
 }
 
+sub INFIX_COMMA_SLIP_HELPER(\reified, \future) {
+    nqp::stmts(
+      (my $list :=
+        nqp::p6bindattrinvres(nqp::create(List),List,'$!reified',reified)),
+      nqp::bindattr($list,List,'$!todo',
+        my $todo:= nqp::create(List::Reifier)),
+      nqp::bindattr($todo,List::Reifier,'$!reified',reified),
+      nqp::bindattr($todo,List::Reifier,'$!future',nqp::getattr(future,List,'$!reified')),
+      nqp::bindattr($todo,List::Reifier,'$!reification-target',reified),
+      $list
+    )
+}
+
 # The , operator produces a List.
 proto sub infix:<,>(|) is pure {*}
 multi sub infix:<,>() { nqp::create(List) }
-multi sub infix:<,>(Int \a, Int \b) is default {
-    nqp::p6bindattrinvres(nqp::create(List),List,'$!reified',nqp::list(a,b))
+multi sub infix:<,>(Slip:D \a, Slip:D \b) {
+    nqp::stmts(
+      (my $reified := nqp::create(IterationBuffer)),
+      # now set up the List with a future
+      INFIX_COMMA_SLIP_HELPER($reified, nqp::list(a,b))
+    )
+}
+multi sub infix:<,>(Any \a, Slip:D \b) {
+    nqp::stmts(  # Slip seen, first copy non-slippy thing
+      (my $reified := nqp::setelems(nqp::create(IterationBuffer),1)),
```

Comment:
I don't think the `setelems` helps here much? It probably stops it rounding up to, I think, 4, which when there's something slipped in after seems like a less than great guess. Sure, it could be `Empty`, but it might not be.